graphblas_algorithms/algorithms/core.py

```diff
@@ -0,0 +1,46 @@
+from graphblas import Matrix, select, agg
+from graphblas.semiring import plus_pair
+
+from graphblas_algorithms.classes.graph import to_undirected_graph, Graph
+from graphblas_algorithms.utils import get_all, not_implemented_for
+
+
+def k_truss_core(G, k):
+    id_to_key = G.id_to_key
+    # Ignore self-edges
+    S = G.get_property("offdiag")
+
+    if k < 3:
+        # Most implementations consider k < 3 invalid,
+        # but networkx leaves the graph unchanged
+        C = S
+    else:
+        # Remove edges not in k-truss
+        nvals_last = S.nvals
+        C = Matrix("int32", S.nrows, S.ncols)
+        while True:
+            C(S.S, replace=True) << plus_pair(S @ S.T)
+            C << select.value(C >= k - 2)
+            if C.nvals == nvals_last:
+                break
+            nvals_last = C.nvals
+            S = C
+
+    # Remove isolate nodes
+    idx, _ = C.reduce_rowwise(agg.count).to_values()
```

Comment:
Why not use `monoid.any` here?

Also, SuiteSparse:GraphBLAS allows `to_values` to return only indices or values by passing in NULL for the results you don't want (this is an extension of the spec).  Should we use this and expose this so that only the indices or values can be obtained?

Also, I prefer the name `indices` here instead of `idx`.  To me, `idx` means a single index.

Reply:
I'll change idx -> indices and use monoid.any. We can worry about the `to_values` extension later. Yes, it's a waste, but is only done once in the algorithm, and only on a Vector. So pretty small effect.